src/components/dialog/createDialog.ts

```diff
@@ -0,0 +1,76 @@
+import { ComposeFactory } from 'dojo-compose/compose';
+import { VNodeProperties } from 'dojo-interfaces/vdom';
+import { DNode, Widget, WidgetOptions, WidgetProperties, WidgetState } from '../../interfaces';
+import createWidgetBase from '../../createWidgetBase';
+import { v } from '../../d';
+
+export interface DialogState extends WidgetState {
+	modal?: boolean;
+	open?: boolean;
+	title?: string;
+	underlay?: boolean;
+};
+
+export interface DialogProperties extends WidgetProperties {
+	modal?: boolean;
+	open?: boolean;
+	title?: string;
+	underlay?: boolean;
+	onOpen?(): void;
+	onRequestClose?(): void;
```

Comment:
I was thinking about the use case (probably rare) where a dialog shouldn't be able to be closed by a user clicking the close button. Maybe an application wants to show a dialog during a network operation and hide it when it's complete. Now that I think about it, we probably want to leave `onRequestClose` as optional, but add a `closeable` boolean option that hides the close button.